### DISCUSSION TITLE
How FastAPI handles path matching?

### DISCUSSION THREAD

**Discussion by seifalmotaz** [fastapi/fastapi#4967] [Questions]
Hi, I am a Dart developer. I am developing a new server-side library something like FastAPI (I inspired a lot of FastAPI).

I am using Regular Expressions to match the request path and the handler path:
example: `/my/path/1567` to match `/my/path/([0-9]+)`

but I was wondering if there is a better way to handle the route path that makes the matching take less time.

Thanks.
My project/library: [https://seifalmotaz.github.io/cruky/](https://seifalmotaz.github.io/cruky/)

**Comment by lnxpy** [comment]
[This official documentation](https://docs.djangoproject.com/en/4.0/topics/http/urls/) shows that how Django handles the URL pattern matching. It used to be RegEx as the URL patterns but now they have `path` as well. Hope you find if useful.
